pkg/cache/memory/repository.go

```diff
@@ -83,13 +83,6 @@ func newRepository(id string, repoSpec *configapi.Repository, repo repository.Re
 	return r
 }
 
-func (r *cachedRepository) Refresh(ctx context.Context) error {
-
```

Comment:
Do you need the mutex on the entire cache? Or do we need a way of refreshing just one repo?